docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,91 @@
 title: Executor
 ---
 
-* Runtime Versioning
+The executor is responsible for dispatching and executing calls into the Substrate runtime.
 
-* Native VS Wasm Execution
+## Runtime Execution
 
-* Forkless Runtime Upgrades
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly (Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when the runtime is upgraded. The executor determines which version of the runtime to use when dispatching and executing calls.
+
+### Execution Strategy
+
+Substrate allows you to configure an execution strategy during different parts of the blockchain execution process. The strategies are:
+
+- `Native`: Execute with native build (if available, WebAssembly otherwise).
+- `Wasm`: Only execute with the WebAssembly build.
+- `Both`: Execute with both native (where available) and WebAssembly builds.
+- `NativeElseWasm`: Execute with the native build if possible; if it fails, then execute with WebAssembly.
+
+The default execution strategies for the different parts of the blockchain execution process are:
+
+- Syncing: `NativeElseWasm`
+- Block Import: `NativeElseWasm`
+- Block Construction: `Wasm`
+- Off-Chain Worker: `Native`
+- Other: `Native`
+
+### Wasm Execution
+
+The Wasm representation of the Substrate runtime is considered the canonical runtime. Because this Wasm runtime is placed in the blockchain storage, the network must come to consensus about this binary. Thus it can be verified to be consistent across all syncing nodes.
+
+The Wasm execution environment is more restrictive than the native execution environment, because it makes few assumptions about the target machine it will run on. For example, the Wasm runtime always executes in a 32-bit environment.
+
+For these reasons, the blockchain prefers to do block construction with the Wasm runtime even though Wasm execution is measurably slower than native execution. Wasm execution can help to ensure that block producers create valid blocks.
+
+### Native Execution
+
+For all other execution processes other than block construction, the native runtime is preferred since it is more performant. However, in any situation where the executor determines the native executable should not be run, the canonical Wasm runtime is chosen instead.
```

Comment:
Hmm, this "determines", I would rewrite this. The executor does not uses the native when it is told or when the version of the native is not compatible with the requested runtime version.

Reply:
Updated